### PR TITLE
remove sympy dependency, add basic unit tests

### PR DESCRIPTION
**Pull request by arcturusannamalai**
- remove sympy dependency
- add a vanilla unittest to be run with python3 -m unittest
- readme example in test

### CODE REVIEW

**Comment by danieleongari** [comment]
@arcturusannamalai Thanks a lot for your contribution, I will review it soon. 
Could you please add a reference to the `isprime` function in the docstring, please. 
To better understand what are the limits and exceptions of such a few-lines function.
Thanks!

**Comment by arcturusannamalai** [comment]
@danieleongari - this 'isprime' is a basic primality checker in a somewhat slow method; it works fine AFAIK <s> - not sure if this basic code needs any kind of docstrings</s>

File: definitive_screening_design/_generalized_dsd.py [hunk 1]
```diff
@@ -32,8 +32,11 @@
 """
 
 import numpy as np
-from sympy import isprime, legendre
 
+def isprime(p):
+    if p <= 2: return True
```
Comment (arcturusannamalai) [inline definitive_screening_design/_generalized_dsd.py:39]:
```suggestion
    """  this 'isprime' is a basic primality checker in a somewhat slow method; it works to replace sympy implementation;
           we check for 1, 2 as prime, even numbers as not-prime, and then see if residue of number @p with divisors from (2 to ceil(p/2)+1) if they are non-zero to ensure primality"""
    if p <= 2: return True    
```

**Review by arcturusannamalai — COMMENTED** [review]
added doc string comments

**Comment by danieleongari** [comment]
Looks good to me.
Thanks a lot for your contibution!

**Pull request merged**